packages/flutter/lib/src/material/tabs.dart

```diff
@@ -398,7 +398,8 @@ class _IndicatorPainter extends CustomPainter {
     if (controller.indexIsChanging) {
       // The user tapped on a tab, the tab controller's animation is running.
       final Rect targetRect = indicatorRect(size, controller.index);
-      _currentRect = Rect.lerp(targetRect, _currentRect ?? targetRect, _indexChangeProgress(controller));
```

Comment:
it was lerping between targetRect and _currentRect, in which _currentRect was getting updated every frame. Thus it moved faster than intended.